File: benchmarks/bench_loop.cpp
// bench_loop.cpp — sum with data-dependent loop, prevents constant folding
#include <iostream>
#include <windows.h>
int main(int argc, char** argv) {
    (void)argv;
    LARGE_INTEGER freq, t0, t1;
    QueryPerformanceFrequency(&freq);
    QueryPerformanceCounter(&t0);
    long long s = 0;
    long long start = (long long)argc;
    for (long long i = start; i <= 100000000LL; i++) s += i;
    QueryPerformanceCounter(&t1);
    double ms = (double)(t1.QuadPart-t0.QuadPart)*1000.0/freq.QuadPart;
    std::cout << s << "\n"; std::cout.flush();
    fprintf(stderr, "COMPUTE_MS=%.3f\n", ms);
    return 0;
}

File: benchmarks/bench_loop.py
import time, sys
t0 = time.perf_counter()
s = 0
for i in range(1, 100000001):
    s += i
ms = (time.perf_counter() - t0) * 1000
print(s)
print(f'COMPUTE_MS={ms:.3f}', file=sys.stderr)
